src/main/java/com/github/snuk87/keycloak/kafka/KafkaStandardProducerFactory.java

```diff
@@ -3,21 +3,25 @@
 import java.util.Map;
 import java.util.Properties;
 
+import com.github.snuk87.keycloak.kafka.serializer.JsonSerializer;
 import org.apache.kafka.clients.producer.KafkaProducer;
 import org.apache.kafka.clients.producer.Producer;
 import org.apache.kafka.clients.producer.ProducerConfig;
-import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public final class KafkaStandardProducerFactory implements KafkaProducerFactory {
 
+	private static final Logger log = LoggerFactory.getLogger(KafkaStandardProducerFactory.class);
+
 	@Override
-	public Producer<String, String> createProducer(String clientId, String bootstrapServer,
+	public Producer<String, Object> createProducer(String clientId, String bootstrapServer,
 			Map<String, Object> optionalProperties) {
 		Properties props = new Properties();
 		props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
 		props.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
-		props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-		props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+		props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, JsonSerializer.class.getName());
```

Comment:
This will be a breaking change for all system that uses plain json strings. I think it would be much better to make this configurable and rely and on the default serializer from the `org.apache.kafka.common.serialization` package.

Reply:
Yes you are right, it was a mistake to change the KEY_SERIALIZER. I fixed it.
In my next PR both will be configurable. Thank you!